### PR TITLE
Pyrakoon

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

File: pylabs/Compat.py [hunk 1]
```diff
@@ -14,6 +14,26 @@ class Status:
...
 
+VAR = 'ARAKOON_PYTHON_CLIENT'
+def _normal_client():
+    logging.info("opting for normal client")
+    global arakoon_client
+    from arakoon import Arakoon
+    arakoon_client = Arakoon
+
+if os.environ.has_key(VAR):
+    global arakoon_client
```
Comment (NicolasT) [inline pylabs/Compat.py:25]: No need for this `global` declaration.

File: pylabs/Compat.py [hunk 2]
```diff
@@ -14,6 +14,26 @@ class Status:
...
+    wanted = os.environ[VAR]
+    if wanted == 'pyrakoon':
+        logging.info("opting for pyrakoon")
+        print "pyrakoon"
+        from pyrakoon import compat
+        arakoon_client = compat
+    else:
+        _normal_client()
```
Comment (NicolasT) [inline pylabs/Compat.py:33]: Why put this case in a function and the other case toplevel?

File: pylabs/test/server/quick/system_tests_basic.py [hunk 3]
```diff
@@ -109,7 +106,13 @@ def test_max_value_size_tinkering ():
     time.sleep(1.0)
     C.assert_running_nodes(1)
     client = C.get_client()
-    assert_raises (ArakoonException, client.set, key, value)
+    try:
+        client.set(key, value)
+        assert_true(False)
+    except Exception, e:
+        # with multiple clients, we have multiple ArakoonException hierarchies
+        assert_equals(e.__class__.__name__, 'ArakoonException')
```
Comment (NicolasT) [inline pylabs/test/server/quick/system_tests_basic.py:114]: :worried:

File: pylabs/test/server/quick/system_tests_basic.py [hunk 4]
```diff
@@ -197,15 +200,14 @@ def test_aSSert_scenario_1():
...
+    assert_raises( X.arakoon_client.ArakoonAssertionFailed, client.aSSert, 'x', None)
 
 @C.with_custom_setup(C.default_setup, C.basic_teardown)
 def test_aSSert_scenario_3():
     client = C.get_client()
     client.set('x','x')
-    ass = arakoon.ArakoonProtocol.Assert('x','x')
-    seq = arakoon.ArakoonProtocol.Sequence()
-    seq.addUpdate(ass)
+    seq = client.makeSequence()
```
Comment (NicolasT) [inline pylabs/test/server/quick/system_tests_basic.py:209]: I guess this shows some API difference between the `arakoon` package and `pyrakoon.compat`, where the latter is supposed to be API-compatible with the former. As such, shouldn't `pyrakoon.compat` be updated somehow?

File: pylabs/test/server/quick/system_tests_basic.py [hunk 5]
```diff
@@ -340,6 +341,7 @@ def test_drop_master_singleton():
         raise Error
     except Exception, e:
         rc = e.args[0]
+        ARA_ERR_NOT_SUPPORTED = 32
```
Comment (NicolasT) [inline pylabs/test/server/quick/system_tests_basic.py:344]: Same as above, I guess this points at some API difference.

Having these magic numbers embedded in some test code might not be the right approach.

File: pylabs/test/server/shaky/test_sso_deployment.py [hunk 6]
```diff
@@ -34,6 +34,7 @@
 
 @Common.with_custom_setup( Common.setup_1_node_forced_master, Common.basic_teardown )
 def test_sso_deployment():
+    pass
```
Comment (NicolasT) [inline pylabs/test/server/shaky/test_sso_deployment.py:37]: Leftover?

File: src/client/python/ArakoonProtocol.py [hunk 7]
```diff
@@ -145,6 +145,13 @@ def getNodeLocations(self, nodeId):
         """
         return self._nodes[ nodeId ]
 
+    def getNodeLocation(self, nodeId):
```
Comment (NicolasT) [inline src/client/python/ArakoonProtocol.py:148]: I'm not sure this is the best place to put this...
- It's not entirely clear what this method does compared to `getNodeLocations` (consider adding a docstring at least)
- Is this method useful from a user perspective?
- It changes API, so the `pyrakoon.compat` API should be updated accordingly.

File: pylabs/test/server/quick/system_tests_basic.py (hunk 4 above)
Comment (toolslive) [inline pylabs/test/server/quick/system_tests_basic.py:209]: it is. (see pull request over there)

File: pylabs/test/server/quick/system_tests_basic.py (hunk 5 above)
Comment (toolslive) [inline pylabs/test/server/quick/system_tests_basic.py:344]: alternative is to expose these constants in both (with a standard name)

File: pylabs/test/server/shaky/test_sso_deployment.py (hunk 6 above)
Comment (toolslive) [inline pylabs/test/server/shaky/test_sso_deployment.py:37]: nope. deliberate. nose will now list the test name iso the doc string.

File: src/client/python/ArakoonProtocol.py (hunk 7 above)
Comment (toolslive) [inline src/client/python/ArakoonProtocol.py:148]: This makes both ClientConfig classes interchangeable for both clients. I'm not sure this is the best approach neither.

File: pylabs/Compat.py (hunk 2 above)
Comment (toolslive) [inline pylabs/Compat.py:33]: I use the function twice. it's just about not repeating code.
(I could reorder the tests as we want _normal_client 3 cases out of 4)

File: pylabs/test/server/quick/system_tests_basic.py (hunk 3 above)
Comment (toolslive) [inline pylabs/test/server/quick/system_tests_basic.py:114]: will fix.

**Pull request merged**